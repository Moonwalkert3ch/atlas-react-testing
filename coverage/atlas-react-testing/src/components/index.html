
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for atlas-react-testing/src/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> atlas-react-testing/src/components</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">78.64% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>151/192</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">86.36% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>19/22</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">46.66% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>7/15</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">78.64% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>151/192</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="CoverArt.tsx"><a href="CoverArt.tsx.html">CoverArt.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="CurrentlyPlaying.tsx"><a href="CurrentlyPlaying.tsx.html">CurrentlyPlaying.tsx</a></td>
	<td data-value="82.6" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 82%"></div><div class="cover-empty" style="width: 18%"></div></div>
	</td>
	<td data-value="82.6" class="pct high">82.6%</td>
	<td data-value="46" class="abs high">38/46</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	<td data-value="82.6" class="pct high">82.6%</td>
	<td data-value="46" class="abs high">38/46</td>
	</tr>

<tr>
	<td class="file low" data-value="Footer.jsx"><a href="Footer.jsx.html">Footer.jsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	</tr>

<tr>
	<td class="file high" data-value="PlayControls.tsx"><a href="PlayControls.tsx.html">PlayControls.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="52" class="abs high">52/52</td>
	<td data-value="92.85" class="pct high">92.85%</td>
	<td data-value="14" class="abs high">13/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="52" class="abs high">52/52</td>
	</tr>

<tr>
	<td class="file high" data-value="PlayListItem.tsx"><a href="PlayListItem.tsx.html">PlayListItem.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="15" class="abs high">15/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="15" class="abs high">15/15</td>
	</tr>

<tr>
	<td class="file low" data-value="Playlist.tsx"><a href="Playlist.tsx.html">Playlist.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="25" class="abs low">0/25</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="25" class="abs low">0/25</td>
	</tr>

<tr>
	<td class="file high" data-value="SongTitle.tsx"><a href="SongTitle.tsx.html">SongTitle.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="VolumeControls.tsx"><a href="VolumeControls.tsx.html">VolumeControls.tsx</a></td>
	<td data-value="88.57" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 88%"></div><div class="cover-empty" style="width: 12%"></div></div>
	</td>
	<td data-value="88.57" class="pct high">88.57%</td>
	<td data-value="35" class="abs high">31/35</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="88.57" class="pct high">88.57%</td>
	<td data-value="35" class="abs high">31/35</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2024-09-23T19:56:11.836Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    