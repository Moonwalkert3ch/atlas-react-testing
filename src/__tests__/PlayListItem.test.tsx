import { render } from '@testing-library/react';
import { expect, test, vi } from 'vitest';
import { PlayListItem } from '../components/PlayListItem';

const onClick = vi.fn();

test('PlayListItem renders correctly when current', () => {
  const { container } = render(
    <PlayListItem 
      title='Painted in Blue' 
      artist='Soul Canvas' 
      duration='5:55' 
      onClick={onClick} 
    />
  );
  expect(container).toMatchSnapshot();
});

test('PlayListItem renders correctly when not current', () => {
  const { container } = render(
    <PlayListItem 
      title='Tidal Drift' 
      artist='Echoes of the Sea' 
      duration='8:02' 
      isCurrent={false} 
      onClick={onClick} 
    />
  );
  expect(container).toMatchSnapshot();
});
