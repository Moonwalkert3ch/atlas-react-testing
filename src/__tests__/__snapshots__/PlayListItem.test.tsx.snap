// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem renders correctly when current 1`] = `
<div>
  <div
    class="flex flex-col justify-center p-4 relative  bg-white hover:bg-light-yellow transition-colors duration-300 cursor-pointer"
  >
    <div
      class="flex flex-col"
    >
      <p
        class="text-sm font-medium text-slate-500"
      >
        Painted in Blue
      </p>
      <p
        class="text-sm font-medium text-black"
      >
        Soul Canvas
      </p>
    </div>
    <p
      class="absolute right-0 top-1/2 transform -translate-y-1/2 text-sm font-medium text-slate-500"
    >
      5:55
    </p>
  </div>
</div>
`;

exports[`PlayListItem renders correctly when not current 1`] = `
<div>
  <div
    class="flex flex-col justify-center p-4 relative  bg-white hover:bg-light-yellow transition-colors duration-300 cursor-pointer"
  >
    <div
      class="flex flex-col"
    >
      <p
        class="text-sm font-medium text-slate-500"
      >
        Tidal Drift
      </p>
      <p
        class="text-sm font-medium text-black"
      >
        Echoes of the Sea
      </p>
    </div>
    <p
      class="absolute right-0 top-1/2 transform -translate-y-1/2 text-sm font-medium text-slate-500"
    >
      8:02
    </p>
  </div>
</div>
`;
